goldens/golden_test.go

```diff
@@ -119,3 +124,21 @@ func showTopLevel(dir string) (string, error) {
 	b, err := exec.Command("git", "-C", dir, "rev-parse", "--show-toplevel").Output()
 	return strings.TrimSpace(string(b)), err
 }
+
+func discoverWantStderr(want string) []string {
+	lines := strings.Split(want, "\n")
+	var wantStderr []string
+	for _, l := range lines {
+		want, ok := strings.CutPrefix(l, "STDERR: ")
+		if ok {
+			wantStderr = append(wantStderr, want)
+		}
+	}
+	return wantStderr
+}
+
+func compareStderr(want, got []string) string {
+	return cmp.Diff(want, got, cmp.Comparer(func(x, y string) bool {
```

Comment:
Suggestion: what about making the variable _just_ the cmp.Option? i.e., just throw the `cmp.Comparer` into a package level var. That way in the test, it's clearer what we're doing (we're calling `cmp.Diff` with a custom option) and it's more extensible if we need more options later. It'll still be pretty compact and this is IMO a very typical way to handle verbose compare opts.